ddtrace/internal/ci/recorder.py

```diff
@@ -0,0 +1,184 @@
+    config = attr.ib(default=None)
+    _tags = attr.ib(type=dict, factory=ci.tags)
+    _service = attr.ib(type=str, default=None)
+    _codeowners = attr.ib(default=None)
+
+    enabled = False
+    _instance = None  # type: ClassVar[Optional[CIRecorder]]
+    _lock = forksafe.Lock()
+
+    def __attrs_post_init__(self):
+        # type: () -> None
+        self.tracer = self.tracer or ddtrace.tracer
+        self.tracer.on_start_span(self._set_test_defaults_on_span)
+
+        # detect service name from repository URL if needed
+        service = int_service(None, self.config)
+        if (
+            # repository URL was detected from environment or .git
+            self._tags.get(ci.git.REPOSITORY_URL, None)
+            # the service name was not changed
+            and self.config
+            and service == self.config._default_service
+        ):
+            repository_name = _extract_repository_name(self._tags[ci.git.REPOSITORY_URL])
+            self._service = repository_name
+        elif self._service is None and service is not None:
+            self._service = service
+
+        try:
+            from ddtrace.internal.codeowners import Codeowners
+
+            self._codeowners = Codeowners()
+        except ValueError:
+            # the CODEOWNERS file is not available
+            pass
+        except Exception:
+            log.warning("Failed to load CODEOWNERS", exc_info=True)
+
+    def _set_test_defaults_on_span(
+        self,
+        span,  # type: Span
+    ):
+        # type: (...) -> None
+        if span.parent_id is None:
+            # DEV: shall we assert that the root span is a test span?
+            # We could remove that check from TraceCiVisibilityFilter.
+            if span.span_type != SpanTypes.TEST:
+                log.warning("Root span is not a test span. Skipping CI Visibility mode.")
+                return
+
+            span.service = self._service
+            span.context.dd_origin = ci.CI_APP_TEST_ORIGIN
+            span.context.sampling_priority = AUTO_KEEP
+            span.set_tags(self._tags)
+
+            # DEV: it might not be necessary to add library_version when using agentless mode
+            span._set_str_tag(ci.LIBRARY_VERSION, ddtrace.__version__)
+
+    @classmethod
+    def disable(cls):
+        # type: () -> None
+        with cls._lock:
+            if cls._instance is None:
+                return
+
+            forksafe.unregister(cls._restart)
+
+            recorder = cls._instance
+            recorder.tracer.deregister_on_start_span(recorder._set_test_defaults_on_span)
+            try:
+                recorder.tracer.shutdown()
+            except Exception:
+                log.warning("Failed to shutdown tracer", exc_info=True)
+
+            # Remove required tracer filters
+            # tracer_filters = recorder.tracer._filters
+            # if any(isinstance(tracer_filter, TraceCiVisibilityFilter) for tracer_filter in tracer_filters):
```

Comment:
why is this commented out?

Reply:
The issue is with the configure method called to "unconfigure" filters from a writer.